### ISSUE TITLE
Better markdown support and new markdown features

### ISSUE THREAD

**Issue by merlinfuchs** [merlinfuchs/embed-generator#55]
Discord is currently rolling out extended support for markdown including headings.

**Comment by the-w41k3r** [comment]
Didn't they remove that feature?

**Comment by merlinfuchs** [comment]
Seems to be back but without masked links

**Comment by RiskyMH** [comment]
> Waiting on either [ItzDerock/discord-markdown-parser#5](https://github.com/ItzDerock/discord-markdown-parser/issues/5) or [brussell98/discord-markdown#42](https://github.com/brussell98/discord-markdown/issues/42) to implement it, because I'm too stupid to get it to work myself ...

I saw this issue while I was trying to get the new markdown support, and ended up doing this to get it to work: 
[`brussell98:discord-markdown@master...RiskyMH:discord-markdown@master`](https://github.com/brussell98/discord-markdown/compare/master...RiskyMH:discord-markdown:master#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346). 

There are prob bugs but this:
* uses updated version of `simple-markdown` package
* adds lists (ordered and unordered)
* headings
* masked link by default (but this isn't necessary as much here)

**Comment by merlinfuchs** [comment]
@RiskyMH Thanks a lot of for making the effort and putting this here! I will look into your changes and see what I can use from that 🥳

**Comment by merlinfuchs** [comment]
Headings and lists are now supported. They are some bugs but I think it's better than no support at all ...

The whole markdown parsing + CSS needs a refactor anyway, creating a separate issue for that
